python/cudf/cudf/core/series.py

```diff
@@ -2970,6 +2970,8 @@ def unique(self):
         dtype: object
         """
         res = self._column.unique()
+        if cudf.get_option("mode.pandas_compatible"):
+            return res.values
         return Series(res, name=self.name)
```

Comment:
Only doing this in pandas-compat mode because this marks the buffer as exposed, and may not work for string/struct types?

Reply:
Yup, will only work for numeric types (int, float..)